File: Python/Program/ProgramStudi.py
# Kelas Program Studi
class ProgramStudi():
    # Atribut Mahasiswa:
    __nama_prodi = ""
    __kode = ""
    # Atribut komponen:
    __sivitas_akademik = []
    __course = []

    # Konstruktor
    def __init__(self, nama_prodi, kode, sivitas_akademik, course):
        self.__nama_prodi = nama_prodi
        self.__kode = kode
        self.__sivitas_akademik = sivitas_akademik.copy()
        self.__course = course.copy()

    # Setter
    def set_nama_prodi(self, nama_prodi):
        self.__nama_prodi = nama_prodi

    def set_kode(self, kode):
        self.__kode = kode

    def set_sivitas_akademik(self, sivitas_akademik):
        self.__sivitas_akademik = sivitas_akademik.copy()

    def set_course(self, course):
        self.__course = course.copy()

    # Getter
    def get_nama_prodi(self):
        return self.__nama_prodi

    def get_kode(self):
        return self.__kode
    
    def get_sivitas_akademik(self):
        return self.__sivitas_akademik
    
    def get_course(self):
        return self.__course

File: CPP/Program/ProgramStudi.cpp
// Import file:
#include "header.hh"

// Empty konstruktor.
ProgramStudi::ProgramStudi()
{
    this->nama_prodi = "";
    this->kode = "";
    this->sivitas_akademik = {};
    this->course = {};
}

// Konstruktor with params.
ProgramStudi::ProgramStudi(string nama_prodi, string kode, vector<SivitasAkademik> sivitas_akademik, vector<Course> course)
{
    this->nama_prodi = nama_prodi;
    this->kode = kode;
    this->sivitas_akademik = sivitas_akademik;
    this->course = course;
}

// Setter
void ProgramStudi::set_nama_prodi(string nama_prodi)
{
    this->nama_prodi = nama_prodi;
}
void ProgramStudi::set_kode(string kode)
{
    this->kode = kode;
}
void ProgramStudi::set_course(vector<Course> course)
{
    this->course = course;
}
void ProgramStudi::set_sivitas_akademik(vector<SivitasAkademik> sivitas_akademik)
{
    this->sivitas_akademik = sivitas_akademik;
}

// Getter
string ProgramStudi::get_nama_prodi()
{
    return this->nama_prodi;
}
string ProgramStudi::get_kode()
{
    return this->kode;
}
vector<SivitasAkademik> ProgramStudi::get_sivitas_akademik()
{
    return this->sivitas_akademik;
}
vector<Course> ProgramStudi::get_course()
{
    return this->course;
}

ProgramStudi::~ProgramStudi()
{
    
}